File: programmers/js_src/12947.js
// 하샤드 수

function solution(x) {
  var answer = true;
  
  const arr = x.toString().split('').map(i => parseInt(i));

  const sum = arr.reduce((a, b) => a + b, 0);
  
  answer = x % sum === 0 ? true : false;
  
  return answer;
}

File: programmers/python/12947.py
# 하샤드 수

def solution(x):
    answer = True
    numList = list(map(int, str(x)))

    if x % sum(numList) == 0:
        answer = True
    else:
        answer = False

    return answer
  
# 다른 풀이
# def solution(n):
#     return n % sum([int(c) for c in str(n)]) == 0